lib/rspec-puppet-facts.rb

```diff
@@ -228,7 +229,12 @@ def self.with_custom_facts(os, facts)
       next if fact[:options][:confine] && !fact[:options][:confine].include?(os)
       next if fact[:options][:exclude] && fact[:options][:exclude].include?(os)
 
-      facts[name] = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
+      # if merge_facts passed, merge supplied facts into facts hash
+      if fact[:options][:merge_facts]
+        facts = fact[:value].respond_to?(:call) ? facts.deep_merge!({name.to_sym => fact[:value].call(os, facts)}) : facts.deep_merge!({name.to_sym => fact[:value]})
+      else
+        facts[name.to_sym] = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
+      end
```

Comment:
I'd use additional variables to avoid duplication and make the actual logic clearer:

```suggestion
      key = name.to_sym
      value = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
      
      if fact[:options][:merge_facts]
        facts = facts.deep_merge!({key => value})
      else
        facts[key] = value
      end
```

Reply:
💯 for this. Much cleaner. Thanks @ekohl